snapcraft/tests/unit/store/test_store_client.py

```diff
@@ -1432,6 +1433,16 @@ def test_conflicting_force(self):
         result = self.client.push_metadata('basic', metadata, True)
         self.assertIsNone(result)
 
+    def test_snap_not_found(self):
+        self._setup_snap()
+        metadata = {'field_ok': 'foo'}
+        raised = self.assertRaises(
+            errors.SnapNotFoundError,
+            self.client.push_metadata, 'notbasic', metadata, False)
```

Comment:
Instead of 'notbasic', it would be clearer to use something like: 'test-unexisting-snap'.